src/wilcoxon.jl

```diff
@@ -179,6 +182,44 @@ function pvalue(x::ApproximateSignedRankTest; tail=:both)
     end
 end
 
+ci(x::ApproximateSignedRankTest, alpha::Real=0.05; tail=:both) = calculate_ci(x.vals, alpha; tail=tail)
+
+# implementation method inspired by these notes: http://www.stat.umn.edu/geyer/old03/5102/notes/rank.pdf
+function calculate_ci(x::AbstractVector, alpha::Real=0.05; tail=:both)
+    check_alpha(alpha)
+
+    if tail == :both
+        c = 1 - alpha
+    else
+        c = 1 - 2 * alpha
+    end
+    n = length(x)
+    m = div(n * (n + 1), 2)
+    k_range = 1:div(m, 2)
+    l = [1 - 2 * psignrank(i, n, true) for i in k_range]
+    k = indmin(abs(l-c))
+    vals = Float64[]
```

Comment:
What would the benefit of that be? It seems like it would occasionally lead to incorrect behavior since the elements of the vals array won't always be the same type as the input array (e.g. If the inputs are all ints, say [1, 2], then we would still want to have the vals array be Float64 not Int)